nix/modules/environment.nix

```diff
@@ -0,0 +1,17 @@
+{ lib, config, pkgs, ... }:
+
+{
+  options.environment = {
+    systemPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+    };
+  };
+
+  config = {
+    environment.etc."profile.d/system-manager-path.sh".source =
+      pkgs.writeShellScript "system-manager-path.sh" ''
+        export PATH=${lib.makeBinPath config.environment.systemPackages}:''${PATH}
```

Comment:
This should point to a static filesystem location or relogin will be required every time  `systemPackages` is changed.

Reply:
Right, that's what I described in the issue actually, we'll put it under `/run`.

Reply:
Also, https://github.com/fish-shell/fish-shell/issues/3665#issuecomment-268327507

Changes to the PATH variable should be in /etc/environment:
```bash
# note no 'export'
PATH="/nix/store/v7dj32cnnl52fic916hpnws6b51il3h0-exa-0.10.1/bin:${PATH}"
```

This way it'll be compatible with all the shells (including non-posix like fish).

Reply:
Playing with PATH globally has always been a nuisance.

Reply:
Fish can accept `/etc/fish/conf.d/*.fish` script as
```fish
set -gx PATH ${lib.makeBinPath config.environment.systemPackages}" $PATH
```

just in case, if we use /etc/profile for most of posix-shells and wanna add up fish as an edge-case.

Reply:
@misuzu: we point to a fixed location now (`/run/system-manager/sw`).